vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/MaxInFlightByWorkflow.java

```diff
@@ -0,0 +1,82 @@
+package ca.on.oicr.gsi.vidarr.server;
+
+import ca.on.oicr.gsi.Pair;
+import ca.on.oicr.gsi.vidarr.BasicType;
+import ca.on.oicr.gsi.vidarr.ConsumableResource;
+import ca.on.oicr.gsi.vidarr.ConsumableResourceResponse;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.prometheus.client.Gauge;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+final class MaxInFlightByWorkflow implements ConsumableResource {
+  private static final class MaxState {
+    private int maximum;
+    private final Set<String> running = ConcurrentHashMap.newKeySet();
+  }
+
+  private static final Gauge currentInFlightCount =
+      Gauge.build(
+              "vidarr_in_flight_per_workflow_current",
+              "The current number of workflows are running.")
+          .labelNames("workflow")
+          .register();
+  private static final Gauge maxInFlightCount =
+      Gauge.build(
+              "vidarr_in_flight_per_workflow_max",
+              "The maximum number of workflows that can be run simultaneously.")
+          .labelNames("workflow")
+          .register();
+  private final Map<String, MaxState> workflows = new ConcurrentHashMap<>();
+
+  @Override
+  public Optional<Pair<String, BasicType>> inputFromUser() {
+    return Optional.empty();
+  }
+
+  @Override
+  public String name() {
+    return null;
+  }
+
+  @Override
+  public void recover(String workflowName, String workflowVersion, String vidarrId) {
+    workflows.computeIfAbsent(workflowName, k -> new MaxState()).running.add(vidarrId);
```

Comment:
Yes, but that shouldn't be possible since the workflow name and version are coming out of the database, so the database constraints should mean this is populated. I can change it to a `get` if you think that's clearer.

Reply:
I didn't originally in case something really weird happened and failing to run with max = 0 is a valid, though unexpected, path and updating the workflow's max in flight is possible while the server is running.

Reply:
Where do the max per workflow values get set during recovery?

Reply:
Ha...ah...that was missing. Added.